stdlib/public/runtime/HeapObject.cpp

```diff
@@ -572,29 +653,31 @@ void swift::swift_deallocObject(HeapObject *object, size_t allocatedSize,
   // atomic decrement (and has the ability to reconstruct
   // allocatedSize and allocatedAlignMask).
   if (object->weakRefCount.getCount() == 1) {
-    swift_slowDealloc(object, allocatedSize, allocatedAlignMask);
+    SWIFT_RT_ENTRY_CALL(swift_slowDealloc)
+         (object, allocatedSize,
+          allocatedAlignMask);
   } else {
-    swift_unownedRelease(object);
+    SWIFT_RT_ENTRY_CALL(swift_unownedRelease)(object);
   }
 }
 
 void swift::swift_weakInit(WeakReference *ref, HeapObject *value) {
   ref->Value = value;
-  swift_unownedRetain(value);
+  SWIFT_RT_ENTRY_CALL(swift_unownedRetain)(value);
 }
 
 void swift::swift_weakAssign(WeakReference *ref, HeapObject *newValue) {
-  swift_unownedRetain(newValue);
+  SWIFT_RT_ENTRY_CALL(swift_unownedRetain)(newValue);
   auto oldValue = ref->Value;
   ref->Value = newValue;
-  swift_unownedRelease(oldValue);
+  SWIFT_RT_ENTRY_CALL(swift_unownedRelease)(oldValue);
 }
 
 HeapObject *swift::swift_weakLoadStrong(WeakReference *ref) {
   auto object = ref->Value;
   if (object == nullptr) return nullptr;
   if (object->refCount.isDeallocating()) {
-    swift_unownedRelease(object);
+    SWIFT_RT_ENTRY_CALL(swift_unownedRelease)(object);
     ref->Value = nullptr;
     return nullptr;
   }
```

Comment:
I think Instruments only needs to intercept the primitive refcounting operations.